### ISSUE TITLE
Here's another lost predicate

### ISSUE THREAD

**Issue by adMartem** [javacc21/javacc21#229]
```
Limpid : 
        FooBarNeverMore =>|| | "baz"
;

FooBarNeverMore :
   ( NonNever =>|| | "never" )  // NonNever SCAN succeeds if "never"! Doesn't check predicate
;

NonNever :
   ( "foo" | 
     "bar" | 
     SCAN {isFalse()}# => "never" |
     "more"
   )
;
```

**Comment by revusky** [comment]
That's kind of an interesting little bug. It's also a bug in this single-token optimization logic, but in particular, it only manifests itself due to the presence of superfluous parentheses. The parentheses in both `FooBarNeverMore` and `NonNever` are superfluous, and if you remove them, the bug is gone!

Granted, you should be able to put as many superfluous parentheses in your code as you feel like. (*De gustibus no es disputandum.*) So it was clearly a bug.

Now fixed, I believe. You can try it. Thanks again for the feedback.

**Comment by adMartem** [comment]
That explains why I had trouble reducing this to a simple test case.  I started with the simple structure I thought would exhibit the bug, but it didn't, so I reduced the actual code incrementally until reaching the state I posted (requisite superfluous parentheses and all).  Some of the legacy superfluous parentheses (maybe these) date back to when I discovered that the behavior of JavaCC (the elder) was sometimes dependent on them.  

[ Interesting word, "parenthesis", as Oxford defines it to be a word or phrase delimited by a pair of indicators (paraphrasing), so the plural would seem to denote more than one pair (as you and I used correctly).  I always have to catch myself when referring to single parenthetical element to keep from using the plural form.]
